File: mosakin-backend/src/main/kotlin/com/mosa/office/kintai/controller/model/PaidList.kt
package com.mosa.office.kintai.controller.model

import java.time.LocalDate

data class PaidListHeaderViewModel(
    val carryForward : Int,
    val annualPaidNumber : Int,
    val leftPaidNumber: Int,
    val currentPaidAcquisitionNumber: Int
)

data class PaidListItemViewModel(
    val paidId : String,
    val paidAcquisitionDate : LocalDate,
    val paidTimeType : String
)

data class PaidListViewModel(val header : PaidListHeaderViewModel,val list : List<PaidListItemViewModel>)

File: mosakin-frontend/src/models/models/paidList.ts
import { DateValue } from "./common";

export interface PaidListViewModel {
  list: PaidListItem[];
  header: PaidListHeader;
}

export interface PaidListHeader {
  carryForward: number;
  annualPaidNumber: number;
  leftPaidNumber: number;
  currentPaidAcquisitionNumber: number;
}

export interface PaidListItem {
  paidId: string;
  paidAcquisitionDate: DateValue;
  paidTimeType: PaidTimeType;
}

export type PaidTimeType = "ALL_DAY" | "AM" | "PM";
